gazebo_ros2_control/src/gazebo_system.cpp

```diff
@@ -0,0 +1,389 @@
+
+  /// \brief The effort command interfaces of the joints
+  std::vector<std::shared_ptr<hardware_interface::CommandInterface>> joint_eff_cmd_;
+
+  /// \brief callback to get the e_stop signal from the ROS 2 topic
+  void eStopCB(const std::shared_ptr<std_msgs::msg::Bool> e_stop_active)
+  {
+    this->e_stop_active_ = e_stop_active->data;
+  }
+};
+
+namespace gazebo_ros2_control
+{
+
+bool GazeboSystem::initSim(
+  rclcpp::Node::SharedPtr & model_nh,
+  gazebo::physics::ModelPtr parent_model,
+  std::vector<transmission_interface::TransmissionInfo> transmissions,
+  sdf::ElementPtr sdf)
+{
+  this->dataPtr = std::make_unique<GazeboSystemPrivate>();
+  this->dataPtr->last_update_sim_time_ros_ = rclcpp::Time();
+
+  this->nh_ = model_nh;
+  this->dataPtr->parent_model_ = parent_model;
+  this->dataPtr->n_dof_ = transmissions.size();
+
+  this->dataPtr->joint_names_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_control_methods_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_position_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_velocity_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_effort_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_pos_state_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_vel_state_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_eff_state_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_position_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->last_joint_position_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_velocity_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_effort_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_pos_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_vel_cmd_.resize(this->dataPtr->n_dof_);
+  this->dataPtr->joint_eff_cmd_.resize(this->dataPtr->n_dof_);
+
+  gazebo::physics::PhysicsEnginePtr physics = gazebo::physics::get_world()->Physics();
+
+  std::string physics_type_ = physics->GetType();
+  if (physics_type_.empty()) {
+    RCLCPP_ERROR(this->nh_->get_logger(), "No physics engine configured in Gazebo.");
+    return false;
+  }
+
+  for (unsigned int j = 0; j < this->dataPtr->n_dof_; j++) {
+    //
+    // Perform some validation on the URDF joint and actuator spec
+    //
+    // Check that this transmission has one joint
+    if (transmissions[j].joints.empty()) {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Transmission " << transmissions[j].name <<
+          " has no associated joints.");
+      continue;
+    } else if (transmissions[j].joints.size() > 1) {
+      RCLCPP_WARN_STREAM(
+        this->nh_->get_logger(), "Transmission " << transmissions[j].name <<
+          " has more than one joint. Currently the default robot hardware simulation " <<
+          " interface only supports one.");
+      continue;
+    }
+    std::string joint_name = this->dataPtr->joint_names_[j] = transmissions[j].joints[0].name;
```

Comment:
you get all this data from the `hardware_info` in the `configure` method. No need for transmissions here.